spec-tests/src/test/java/org/candlepin/spec/pools/RefreshPoolsTest.java

```diff
@@ -0,0 +1,883 @@
+    }
+
+    @Test
+    @OnlyInHosted
+    public void shouldRegenerateEntitlementsWhenBrandingForAProductOfAnEntitledPoolChanges() {
+        BrandingDTO brand = Branding.build(StringUtil.random("brand-"), StringUtil.random("name-"))
+            .productId(StringUtil.random("prod-"));
+        ProductDTO prod = adminClient.hosted().createProduct(Products.random().branding(Set.of(brand)));
+        adminClient.hosted().createSubscription(Subscriptions.random(owner, prod));
+
+        refreshPools(adminClient, owner.getKey());
+        List<PoolDTO> pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+        PoolDTO pool = pools.get(0);
+
+        // Verify the product exists in its initial state
+        ProductDTO actualProd = adminClient.ownerProducts().getProductByOwner(ownerKey, prod.getId());
+        assertNotNull(actualProd);
+        compareProducts(prod, actualProd);
+        compareBrandings(prod.getBranding(), actualProd.getBranding());
+
+        // Consume the pool so we have an entitlement
+        ConsumerDTO user = Consumers.random(owner, ConsumerTypes.System);
+        user.setUsername("admin");
+        user = adminClient.consumers().createConsumer(user, "admin",
+        ownerKey, null, true);
+        ApiClient consumerClient = ApiClients.trustedConsumer(user.getUuid());
+
+        JsonNode ents = consumerClient.consumers().bindPool(user.getUuid(), pool.getId(), 1);
+        assertEquals(1, ents.size());
+        JsonNode ent = ents.get(0);
+        JsonNode cert = ent.get("certificates").get(0);
+        JsonNode serial = cert.get("serial").get("serial");
+
+        user = adminClient.consumers().getConsumer(user.getUuid());
+        assertEquals(1, user.getEntitlementCount());
+
+        // Update the name of the branding of the upstream product...
+        brand.setName(StringUtil.random("brand-"));
+        prod.setBranding(Set.of(brand));
+        prod = adminClient.hosted().updateProduct(prod.getId(), prod);
+
+        refreshPools(adminClient, owner.getKey());
+        pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+
+        // Verify the branding change on the product has been pulled down
+        actualProd = adminClient.ownerProducts().getProductByOwner(ownerKey, prod.getId());
+        assertNotNull(actualProd);
+        compareProducts(prod, actualProd);
+        compareBrandings(prod.getBranding(), actualProd.getBranding());
+
+        // Verify the entitlement cert has changed as a result
+        EntitlementDTO updatedEnt = adminClient.entitlements().getEntitlement(ent.get("id").asText());
+        assertThat(updatedEnt.getCertificates())
+            .hasSize(1)
+            .map(CertificateDTO::getSerial)
+            .isNotNull()
+            .map(CertificateSerialDTO::getSerial)
+            .isNotNull()
+            .isNotEqualTo(serial.asLong());
```

Comment:
This test chain isn't performing as you'd expect from the naming of the `map` function. Generally speaking, `extracting` and `map` off a IterableAssert map the whole collection using the mapping function provided, leaving you with another collection as the target under test.

The end result of this by the time you reach `.isNotEqualTo` is a single-element collection containing the serial. As such, the collection will never be equal to the target serial.

To do what you intend here, we need to use `.singleElement` to get out of the collection context, and then from there we can use `.extracting` to map to the desired type:

```
assertThat(updatedEnt.getCertificates())
    .singleElement()
    .isNotNull()
    .extracting(CertificateDTO::getSerial)
    .isNotNull()
    .extracting(CertificateSerialDTO::getSerial)
    .isNotEqualTo(serial.asLong());
```

Reply:
Oh, I see what you are saying. Updated